Sources/ArcGISToolkit/Components/FloorFilter/FloorFilter.swift

```diff
@@ -176,5 +176,29 @@ public struct FloorFilter: View {
         .frame(minHeight: 100)
         .environmentObject(viewModel)
         .disabled(viewModel.isLoading)
+        .onChange(of: selection?.wrappedValue) { newValue in
+            // Prevent a double-set if the view model triggered the original change.
+            guard newValue != viewModel.selection else { return }
+            switch newValue {
+            case .site(let site): viewModel.setSite(site, zoomTo: true)
+            case .facility(let facility): viewModel.setFacility(facility, zoomTo: true)
+            case .level(let level): viewModel.setLevel(level)
```

Comment:
One thing that's unclear to me: should the viewpoint be updated automatically when the user updates the selection? It seems like yes but I think that'd be up for debate. It could also be configurable through another modifier.

Reply:
Q1) Viewpoint update: I think we don't need to update the viewpoint automatically until it becomes clear if viewpoint automatic update is common behavior.
Q2) I think it will be rather annoying if the site and facility are auto expanded. I think keeping them closed is an acceptable behavior, but the values should be auto updated if they are already expanded.

Reply:
Great, thank you for that feedback @mpEsri!